### PR TITLE
Separate species clothing overrides into per-slot files.

### LINKED ISSUE #4797: Monkey tails overlap suit sprite backview
[BUG]

**Describe the bug**
When you are a monkey and wear a space suit the tail is covered on all other views besides the back view.

**To Reproduce**
1. Be Monkey
2. Put on space suit
3. Turn to backview.
4. Notice how the tail is no longer covered

**Expected behavior**
Tail covered on all views

**Screenshots**
image

**Additional context**
Definitely an issue with tail sprites overlapping suit sprites.

### PR DESCRIPTION
**Pull request by pgmzeta**
[HELP WANTED] [CLEANLINESS]

## About the PR 
Changes per-species clothing overrides (e.g. monkey, lizard) from a single icon file for each species to an icon file per-slot-per-species. 

Makes tails responsible for their own clothing sprites, so you can put a monkey tail on a saurian and it'll show the correct suit sprites.

Sprite-sheet separation done by @DimWhat 

## Why's this needed? 
Fewer merge conflicts for species clothing; clearer place to put overrides without having to adjust icon state names. (e.g. no more `js_suitname`).
Fixes #4797

## TODOs
- [x] Figure out `tail.dmi` implementation
- [x] Ensure sprite files are up to date just before merge
- [x] Remove old combined sprite files pre-merge

### CODE REVIEW

**Comment by pgmzeta** [comment]
thoughts on having the tail sprites separated:
1. we need to read the tail species type to know what icon to pull from
2. having a monkey tail on a non-monkey species probably needs different offsets (?)

File: code/mob/living/carbon/human/procs/update_icon.dm [hunk 1]
```diff
@@ -303,21 +319,50 @@
...
 		else
 			src.wear_suit.wear_image.icon = src.wear_suit.wear_image_icon
-			src.wear_suit.wear_image.icon_state = wear_state
+		src.wear_suit.wear_image.icon_state = wear_state
 
 		src.wear_suit.update_wear_image(src, src.wear_suit.wear_image.icon != src.wear_suit.wear_image_icon)
 		src.wear_suit.wear_image.color = src.wear_suit.color
 		src.wear_suit.wear_image.alpha = src.wear_suit.alpha
+
+		if (src.organHolder?.tail)
```
Comment (pgmzeta) [inline code/mob/living/carbon/human/procs/update_icon.dm:338]: This needs to be a proc, but not sure where is best to put it.

**Comment by ZeWaka** [comment]
@flrsh

File: code/obj/item/organs/tail.dm [hunk 2]
```diff
@@ -20,6 +20,7 @@
 	var/failure_ability = "clumsy"	// The organ failure ability associated with this organ.
 	var/human_getting_monkeytail = 0	// If a human's getting a monkey tail
 	var/monkey_getting_humantail = 0	// If a monkey's getting a human tail
+	var/clothing_image_icon = null // if the tail has clothing sprites, set this to the appropriate icon
```
Comment (ZeWaka) [inline code/obj/item/organs/tail.dm:23]: change this to codedoc and if you want to change the rest

File: code/datums/mutantraces.dm [hunk 3]
```diff
@@ -100,6 +98,19 @@ TYPEINFO(/datum/mutantrace)
 	/// If our mutant has a female variant that has different organs, these will be used instead
 	var/list/mutant_organs_f = null
 
+	/// icon definitions for mutantrace clothing variants. one icon file per slot.
```
Comment (ZeWaka) [inline code/datums/mutantraces.dm:101]:
```suggestion
	// icon definitions for mutantrace clothing variants. one icon file per slot.
```

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/mob/living/carbon/human/procs/update_icon.dm (hunk 1 above)
Comment (pgmzeta) [reply code/mob/living/carbon/human/procs/update_icon.dm]: Added as `update_tail_clothing` in [19d9f1e](https://github.com/goonstation/goonstation/pull/9773/commits/19d9f1ec4d6fdc72dc4fb93a428c9a2b2de021c1)

File: code/obj/item/organs/tail.dm (hunk 2 above)
Comment (pgmzeta) [reply code/obj/item/organs/tail.dm]: Added in [0ca96b9](https://github.com/goonstation/goonstation/pull/9773/commits/0ca96b9e2da60a73de054e02e5e442837ff98a2e)

File: code/datums/mutantraces.dm (hunk 3 above)
Comment (pgmzeta) [reply code/datums/mutantraces.dm]: Removed in [0ca96b9](https://github.com/goonstation/goonstation/pull/9773/commits/0ca96b9e2da60a73de054e02e5e442837ff98a2e)

**Review by ZeWaka — APPROVED** [review]

**Comment by ZeWaka** [comment]
✅

**Comment by frawhst** [comment]
github actions betrayal

**Pull request merged**